src/languageservice/serviceclient.ts

```diff
@@ -132,7 +134,9 @@ export default class SqlToolsServiceClient {
             _channel = window.createOutputChannel(Constants.serviceInitializingOutputChannelName);
             let logger = new Logger(text => _channel.append(text));
             let serverStatusView = new ServerStatusView();
-            let downloadProvider = new ServiceDownloadProvider(config, logger, serverStatusView);
+            let httpClient = new HttpClient();
+            let decompressProvider = new DecompressProvider();
```

Comment:
I think it's fine. it's lightweight and I need to pass everything in the constructor for unittests

---
In reply to: [93142367](https://github.com/Microsoft/vscode-mssql/pull/557#discussion_r93142367) [](ancestors = 93142367)